### PR TITLE
Added some Rust wrappers around C types

### PR DESCRIPTION
**Pull request by stevenengler**
This implements some Rust-equivalent types for some C types. There is still more to do in future PRs, but this is a start.

### CODE REVIEW

File: src/main/host/descriptor/pipe.rs [hunk 1]
```diff
@@ -159,20 +150,16 @@ impl PipeFile {
         // remove any flags that aren't relevant
         let status = self.filter_status(status);
 
-        if do_set_bits {
-            self.status |= status;
-        } else {
-            self.status &= !status;
-        }
+        self.status.set(status, do_set_bits);
```
Comment (sporksmith) [inline src/main/host/descriptor/pipe.rs:153]: Doesn't this still need to clear the bits if `do_set_bits` is `false`?

File: src/main/host/descriptor/pipe.rs [hunk 2]
```diff
@@ -247,40 +226,36 @@ impl SharedBuf {
...
         event_queue: &mut EventQueue,
     ) {
         let old_status = self.status;
 
-        if do_set_bits {
-            self.status |= status;
-        } else {
-            self.status &= !status;
-        }
+        self.status.set(status, do_set_bits);
```
Comment (sporksmith) [inline src/main/host/descriptor/pipe.rs:249]: Clear if `!do_set_bits`?

**Review by sporksmith — APPROVED** [review]

File: src/main/host/descriptor/pipe.rs (hunk 1 above)
Comment (stevenengler) [reply src/main/host/descriptor/pipe.rs:153]: The documentation for [`set(flags, bool)`](https://docs.rs/bitflags/1.2.1/bitflags/example_generated/struct.Flags.html#method.set) says "Inserts or removes the specified flags depending on the passed value", and I double checked the [source](https://docs.rs/bitflags/1.2.1/src/bitflags/lib.rs.html#698) and it calls `remove(flags)` if it's `false`.

Comment (sporksmith) [reply src/main/host/descriptor/pipe.rs:154]: Ahh, I missed that `do_set_bits` was getting passed as a parameter there

**Pull request merged**